corehq/apps/userreports/reports/filters/values.py

```diff
@@ -106,10 +106,9 @@ def to_sql_values(self):
 
     def _offset_enddate(self, enddate):
         # offset enddate for SQL query
-        # if it is datetime.date object it should not be offset
-        # if it is datetime.datetime object it should be offset to last microsecond of the day
-        if enddate and type(enddate) is datetime.datetime:
-            enddate = enddate + datetime.timedelta(days=1) - datetime.timedelta.resolution
```

Comment:
If I am getting it right.  `enddate + datetime.timedelta(days=1) - datetime.timedelta.resolution` was incrementing enddate by `1` day there check at line `111` was made. 
Replaced this inclusion with `datetime.datetime.combine(enddate, datetime.datetime.max.time())` which always gives end of that day whether its date or datetime. So therefore no need of check

Reply:
Would this not break the [bug](http://manage.dimagi.com/default.asp?230388), since this change is reverting the fix that was made to address that bug?